tests/experimental/optimizer/test_fusion.py

```diff
@@ -41,33 +41,55 @@ def fusion1(app, x, y):
     return 1.0 / (1.0 + app.exp(x - y))
 
 
-def fuse_ga(app, r: GraphArray) -> GraphArray:
-    result_graphs = np.empty_like(r.graphs, dtype=r.graphs.dtype)
-    t = time.time()
-    for grid_entry in r.grid.get_entry_iterator():
-        graph = r.graphs[grid_entry]
-        result_graphs[grid_entry] = FuseGraph(graph, app.km)()
-    print("fusion time", time.time() - t)
-    return GraphArray(r.grid.copy(), r.cluster_state, result_graphs, r.km)
+def fusion2(app, x, y):
+    return x @ y
+
+
+def fusion3(app, s, q, p):
+    return s * (q @ p.T)
 
 
-def ga_op(app, func, x: BlockArray, y: BlockArray, copy_on_op=True) -> BlockArray:
+def ga_op(
+    app, func, x: BlockArray, y: BlockArray, copy_on_op=True, max_args=2
+) -> BlockArray:
     cluster_state: ClusterState = ClusterState(x.km.devices())
     x_ga: GraphArray = GraphArray.from_ba(x, cluster_state, copy_on_op=copy_on_op)
     y_ga: GraphArray = GraphArray.from_ba(y, cluster_state, copy_on_op=copy_on_op)
     op_ga: GraphArray = func(app, x_ga, y_ga)
-    fused_ga: GraphArray = fuse_ga(app, op_ga)
-    t = time.time()
+    start_time = time.time()
+    fused_ga: GraphArray = op_ga.compile_fuse_ga(max_args)
+    end_time = time.time()
     result_ga: GraphArray = RandomTS(
         seed=conftest.rs,
         max_samples_per_step=1,
         max_reduction_pairs=1,
         force_final_action=True,
     ).solve(fused_ga)
-    print("tree search time", time.time() - t)
+
     return BlockArray(result_ga.grid, x.km, result_ga.to_blocks())
 
 
+def ga_op_sparse(
```

Comment:
Can we rename this to `ga_op_sampled_dense_dense`? This name is closer to what's going on in the function body.